components/playback/controls/player/ButtonWidget.vue

```diff
@@ -0,0 +1,199 @@
+<template>
+  <div>
+    <div class="div__button-background" :style="background_cssprops"></div>
+    <span
+      v-for="btn_index in num_of_btn"
+      :key="btn_index"
+      :ref="btn_index.toString()"
+      class="span__button_label"
+      :style="btn_stateprop(btn_index)"
+      @click="selected(btn_index)"
+      @mouseenter="hover_active(btn_index)"
+      @mouseleave="hover_inactive(btn_index)"
+      >{{ button_names[btn_index - 1] }}
+    </span>
+    <div v-if="num_of_verticalline >= 1">
+      <canvas
+        v-for="line_index in num_of_verticalline"
+        :key="line_index"
+        class="canvas__vertical-line"
+        :style="btn_divider_display(line_index)"
+      >
+      </canvas>
+    </div>
+    <canvas
+      class="canvas__common-horizontal-line"
+      :style="'width: ' + (button_widget_width - 10) + 'px;'"
+    >
+    </canvas>
+  </div>
+</template>
+<script>
+export default {
+  name: "ButtonWidget",
+  props: {
+    button_names: {
+      type: Array,
+      required: true,
+    },
+    enabled_color: { type: String, default: "" },
+    disabled_color: { type: String, default: "" },
+    hover_color: {
+      type: Array,
+      required: true,
+    },
+    is_enabled: {
+      type: Array,
+      default: function () {
+        return new Array(this.num_of_btn).fill(true);
```

Comment:
I'm not sure this is working the way you hoped.   When I rendered a component without the `is_enabled` prop supplied, only the first button was enabled.  Please create a unit test replicating this issue that fails, and then proceed to fix it. My suggestion would be to wait until `mounted` to populate the `is_enabled` array if the array was not provided in full as a prop.
![image](https://user-images.githubusercontent.com/5436635/96634123-0cb78b00-12e8-11eb-948e-7074453799c1.png)

Reply:
Yes, rightly identified have fixed the issue with two items
a) There is a JEST testcase in which the prop is_enabled and verified the working failing
b) Then the correction was required on line 48 as return new Array(this.button_names.length).fill(true).
     So now without a mounted hook the is_enabled Array got populated to true, as button_names is required, the
     above mentioned problem is resolved.